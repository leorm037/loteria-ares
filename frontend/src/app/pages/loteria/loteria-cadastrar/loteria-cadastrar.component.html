<form class="needs-validation was-validated" novalidate>
    <div class="card shadow-sm">
        <div class="card-header">
            <h1 class="h4 m-0"><i class="fa-solid fa-clover icon-rotate-45"></i>Nova loteria</h1>
        </div>

        <div class="card-body">
            <div class="row">
                <div class="col-12">
                    <label for="nomeInput" class="form-label">Nome</label>
                    <div class="input-group has-validation">
                        <span class="input-group-text" id="nomeInputGroup"><i class="bi bi-alphabet"></i></span>
                        <input type="text" class="form-control" id="nomeInput" required aria-label="Nome"
                            aria-describedby="Nome completo da loteria.">
                            <div class="invalid-feedback">Campo inv√°lido</div>
                        </div>
                    <div class="form-text">Coloque o nome completo da loteria.</div>
                </div>
            </div>
        </div>

        <div class="card-footer bg-gradient d-flex justify-content-between">
            <div>
                <a data-loading="true" class="btn btn-outline-secondary bg-gradient btn-sm shadow" href="#"><i
                        class="bi bi-arrow-left"></i> Voltar</a>
            </div>
            <div>
                <button type="reset" class="btn btn-secondary bg-gradient btn-sm shadow me-1"><i
                        class="fa-solid fa-broom"></i> Limpar</button>
                <button type="submit" class="btn btn-primary bg-gradient btn-sm shadow"><i class="bi bi-floppy"></i>
                    Salvar</button>
            </div>
        </div>
    </div>
</form>