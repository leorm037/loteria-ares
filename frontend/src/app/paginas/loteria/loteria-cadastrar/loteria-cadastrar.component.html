<p>loteria-cadastrar works!</p>
