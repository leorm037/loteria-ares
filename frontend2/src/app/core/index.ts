// Guards


// Interceptors


// Models
export type { AppConfig } from './models/app-config.model';
export type { UserAuth } from './models/user-auth.model';
export type { UserToken } from './models/user-token.model';


// Services
export { AuthService } from './services/auth.service';
export { TokenService } from './services/token.service';

// Tokens
export * from './tokens';