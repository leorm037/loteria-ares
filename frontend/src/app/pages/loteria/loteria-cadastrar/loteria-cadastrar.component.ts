import { Component, OnInit } from '@angular/core';
import { Breadcrumb } from '../../../interfaces/breadcrumb';
import { BreadcrumbService } from '../../../services/breadcrumb.service';

@Component({
  selector: 'app-loteria-cadastrar',
  imports: [],
  templateUrl: './loteria-cadastrar.component.html'
})
export class LoteriaCadastrarComponent implements OnInit{

  private readonly breadcrumbs: Breadcrumb[] = [
    {
      href: "/loteria",
      iconClass: "fa-solid fa-clover icon-rotate-45",
      texto: "Loteria"
    },
    {
      iconClass: "bi bi-plus",
      texto: "Novo"
    }
  ];

  constructor(
    private breadcrumbService: BreadcrumbService
  ) {}

  ngOnInit(): void {
    this.breadcrumbService.sendBreadcrumb(this.breadcrumbs);
  }

}
