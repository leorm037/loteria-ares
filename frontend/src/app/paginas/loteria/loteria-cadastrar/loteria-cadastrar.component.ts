import { Component } from '@angular/core';

@Component({
  selector: 'app-loteria-cadastrar',
  imports: [],
  templateUrl: './loteria-cadastrar.component.html',
  styleUrl: './loteria-cadastrar.component.css'
})
export class LoteriaCadastrarComponent {

}
