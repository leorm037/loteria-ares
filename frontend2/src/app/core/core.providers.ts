import { Provider } from "@angular/core";
import { environment } from "../../environments/environment.development";
import { APP_CONFIG, API_BASE_URL, IS_PRODUCTION, LOGGER_FN } from "./tokens";

const version = environment.version;
const api_base_url = environment.api_base_url;
const is_production = environment.is_production;

export const CORE_PROVIDERS: Provider[] = [
    {
        provide: APP_CONFIG,
        useValue: {
            version: version,
            api_base_url: api_base_url,
            production: is_production
        }
    },
    {
        provide: API_BASE_URL,
        useValue: api_base_url
    },
    {
        provide: IS_PRODUCTION,
        useValue: is_production
    },
    {
        provide: LOGGER_FN,
        useFactory: () => {
            if (is_production) {
                return () => { };
            } else {
                return (message: string) => {
                    console.log(message);
                };
            }
        }
    }
];