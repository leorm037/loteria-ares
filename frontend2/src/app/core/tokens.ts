import { InjectionToken } from '@angular/core';
import { AppConfig } from './models/app-config.model';

export const APP_CONFIG = new InjectionToken<AppConfig>('APP_CONFIG');

export const API_BASE_URL = new InjectionToken<string>('API_BASE_URL');

export const IS_PRODUCTION = new InjectionToken<boolean>('IS_PRODUCTION');

export const LOGGER_FN = new InjectionToken<(message: string) => void>('LOGGER_FN');