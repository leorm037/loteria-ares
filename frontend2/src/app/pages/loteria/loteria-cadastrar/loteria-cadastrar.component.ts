import { Component, OnInit } from '@angular/core';
import { Breadcrumb } from '../../../interfaces/breadcrumb';
import { BreadcrumbService } from '../../../services/breadcrumb.service';
import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
import { Router } from '@angular/router';

@Component({
  selector: 'app-loteria-cadastrar',
  imports: [ReactiveFormsModule],
  templateUrl: './loteria-cadastrar.component.html'
})
export class LoteriaCadastrarComponent implements OnInit{

  private readonly breadcrumbs: Breadcrumb[] = [
    {
      href: "/loteria",
      iconClass: "fa-solid fa-clover icon-rotate-45",
      texto: "Loteria"
    },
    {
      iconClass: "bi bi-plus",
      texto: "Novo"
    }
  ];

  public form!: FormGroup;

  constructor(
    private breadcrumbService: BreadcrumbService,
    private formBuilder: FormBuilder,
    private router: Router
  ) {}

  ngOnInit(): void {
    this.form = this.formBuilder.group({
      nome: [null, [Validators.required]],
      abreviacao: [null],
      apiUrl: [null, [Validators.required]],
      apostas: [null, [Validators.required]],
      dezenas: [null, [Validators.required]]
    });
    this.breadcrumbService.sendBreadcrumb(this.breadcrumbs);
  }

  public submit() {
    if (this.form.valid) {
      this.router.navigateByUrl('/loteria');
    }
  }

}
