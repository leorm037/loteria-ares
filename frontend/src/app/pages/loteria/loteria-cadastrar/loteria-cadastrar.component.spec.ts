import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LoteriaCadastrarComponent } from './loteria-cadastrar.component';

describe('LoteriaCadastrarComponent', () => {
  let component: LoteriaCadastrarComponent;
  let fixture: ComponentFixture<LoteriaCadastrarComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [LoteriaCadastrarComponent]
    })
    .compileComponents();

    fixture = TestBed.createComponent(LoteriaCadastrarComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
