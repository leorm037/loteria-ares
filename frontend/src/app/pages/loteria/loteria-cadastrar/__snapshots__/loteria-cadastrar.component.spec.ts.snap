// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoteriaCadastrarComponent Deve conter o breadcrumb correto. 1`] = `
Observable {
  "source": BehaviorSubject {
    "_value": [
      {
        "href": "/loteria",
        "iconClass": "fa-solid fa-clover icon-rotate-45",
        "texto": "Loteria",
      },
      {
        "iconClass": "bi bi-plus",
        "texto": "Novo",
      },
    ],
    "closed": false,
    "currentObservers": [],
    "hasError": false,
    "isStopped": false,
    "observers": [],
    "thrownError": null,
  },
}
`;
